qutip/cy/mcsolve.pyx

```diff
@@ -32,15 +32,15 @@ cdef np.ndarray[complex, ndim=1] normalize(complex[::1] psi):
 
 cdef class CyMcOde:
     cdef:
-        int steady_state, store_states, col_args
+        int steady_state, store_states, col_args, store_final_state
         int norm_steps, l_vec, num_ops
         double norm_t_tol, norm_tol
         list collapses
         list collapses_args
         list c_ops
         list n_ops
-        complex[:,::1] states_out
-        complex[:,::1] ss_out
+        object states_out
+        object ss_out
```

Comment:
This feels like it might possibly have some knock-on performance issues due to less perfect indexing.  I doubt it _will_, because these aren't inner loop variables, but could we check?  I also couldn't immediately see why they needed to be changed at all, but that may just be me missing things.

Reply:
Inside loop I used `cdef complex[:,::1] states_out = self.states_out` to ensure no impact on performance.

I changed it to keep the numpy array alive. We create it with `self.states_out = np.zeros(...)` in a function and do `np.array(self.states_out)` in another and I am not sure it is a safe usage for reference counting. There haven't been any issues, but I preferred to use only one `ndarray` just in case. 

Also `np.array(self.states_out)` don't work on not initialized memoryview, so we would have to branch the `return`, but as an object, it is `None` when not used. (Which I should have mode explicit...)

Reply:
I'm surprised that assigning to the memoryview doesn't keep the numpy array alive - I thought the base object got stored within the view, to avoid problems with that (similar to how numpy views keep references to their underlying objects to prevent the memory going stale).

Either way, I'm not actually bothered about it, since as you say, there shouldn't be any performance issues.